dags/atd_knack_employees_tpw_hire.py

```diff
@@ -0,0 +1,122 @@
+# test locally with: docker compose run --rm airflow-cli dags test atd_knack_employees_tpw_hire
+
+import os
+
+from airflow.models import DAG
+from airflow.operators.docker_operator import DockerOperator
+from pendulum import datetime, duration, now
+
+from utils.onepassword import get_env_vars_task
+from utils.slack_operator import task_fail_slack_alert
+from utils.knack import get_date_filter_arg
+
+DEPLOYMENT_ENVIRONMENT = os.getenv("ENVIRONMENT", "development")
+
+DEFAULT_ARGS = {
+    "owner": "airflow",
+    "depends_on_past": False,
+    "start_date": datetime(2015, 1, 1, tz="America/Chicago"),
+    "email_on_failure": False,
+    "email_on_retry": False,
+    "retries": 0,
+    "execution_timeout": duration(minutes=15),
+    "on_failure_callback": task_fail_slack_alert,
+}
+
+REQURED_SECRETS_KNACK_SERVICES = {
+    "PGREST_ENDPOINT": {
+        "opitem": "atd-knack-services PostgREST",
+        "opfield": "production.endpoint",
+    },
+    "PGREST_JWT": {
+        "opitem": "atd-knack-services PostgREST",
+        "opfield": "production.jwt",
+    },
+}
+
+REQUIRED_SECRETS_HR_MANAGER = {
+    "KNACK_APP_ID": {
+        "opitem": "Knack Human Resources (HR)",
+        "opfield": f"production.appId",
+    },
+    "KNACK_API_KEY": {
+        "opitem": "Knack Human Resources (HR)",
+        "opfield": f"production.apiKey",
+    },
+}
+
+REQUIRED_SECRETS_TPW_HIRE = {
+    "KNACK_APP_ID": {
+        "opitem": "TPW Hire Knack App",
+        "opfield": f"{DEPLOYMENT_ENVIRONMENT}.appId",
+    },
+    "KNACK_API_KEY": {
+        "opitem": "TPW Hire Knack App",
+        "opfield": f"{DEPLOYMENT_ENVIRONMENT}.apiKey",
+    },
+}
+
+REQUIRED_SECRETS_KNACK_TO_KNACK = {
+    "KNACK_APP_ID_DEST": REQUIRED_SECRETS_TPW_HIRE["KNACK_APP_ID"],
+    "KNACK_API_KEY_DEST": REQUIRED_SECRETS_TPW_HIRE["KNACK_API_KEY"],
+    "KNACK_APP_ID_SRC": REQUIRED_SECRETS_HR_MANAGER["KNACK_APP_ID"],
+    "KNACK_API_KEY_SRC": REQUIRED_SECRETS_HR_MANAGER["KNACK_API_KEY"],
+}
+
+REQUIRED_SECRETS_HR_MANAGER.update(REQURED_SECRETS_KNACK_SERVICES)
+REQUIRED_SECRETS_TPW_HIRE.update(REQURED_SECRETS_KNACK_SERVICES)
+REQUIRED_SECRETS_KNACK_TO_KNACK.update(REQURED_SECRETS_KNACK_SERVICES)
+
+with DAG(
+    dag_id=f"atd_knack_employees_tpw_hire",
+    description="Copies Banner TPW employee data from the HR knack app to TPW hire knack app.",
+    default_args=DEFAULT_ARGS,
+    schedule_interval="30 1 * * *" if DEPLOYMENT_ENVIRONMENT == "production" else None,
+    tags=["repo:atd-knack-services", "knack", "hr-manager", "tpw-hire", "banner"],
+    catchup=False,
+) as dag:
+    docker_image = "atddocker/atd-knack-services:production"
+    app_name_src =  "hr-manager" 
+    container_src = "view_684"
+    app_name_dest = "tpw-hire"
+    container_dest = "view_148"
```

Comment:
It feels a little like a anti-pattern to have the configuration in the orchestration configuration, but then again, here's as good as anywhere. It is where the program is kicked off from after all. I mostly wanted to flag it since it's a bit of a departure from what we've done in the past -- do you think 1PW is maybe a better place for these? I'm genuinely not sure and curious about your thoughts.

Reply:
at lot of our knack dags follow this pattern, with the app_name and the container as vars. Its the pattern from airflow 1 too, so when the dags were brought over, they brought that pattern with them.

Reply:
yeah, i'm not sure where else to stick these generic CLI args. it would be kinda nice if we could easily look to like one big config file that has all of these across all knack services DAGs 🤷

Reply:
+1 what Chia said. If we want to start a new pattern we should probably refactor all of the knack DAGs in a separate PR. To me it doesn't feel like a 1PW thing because it's just part of the params to the command to the docker image but I don't have strong feelings either way